packages/flutter_test/lib/src/finders.dart

```diff
@@ -498,6 +498,21 @@ class _TextFinder extends MatchFinder {
     } else if (candidate.widget is EditableText) {
       final EditableText editable = candidate.widget;
       return editable.controller.text == text;
+    } else if (candidate.widget is RichText) {
+      bool parentIsText = false;
+      // check if the direct parent is a Text widget. if so, do not match it
+      // twice. We make an exception for semantics since those may be placed
+      // between a Text and Rich text.
+      candidate.visitAncestorElements((Element parent) {
+        if (parent.widget is Text) {
+          parentIsText = true;
+          return false;
+        } else if (parent.widget is ExcludeSemantics || parent.widget is Semantics)
+          return true;
```

Comment:
nit: when you use `{ }` in at least one branch of the `if/else` chain, use `{ }` in all branches (also, IIRC, we transitioned to using curly braces everywhere).